File: app/src/main/java/wangdaye/com/geometricweather/daily/adapter/model/DailyAstro.java
package wangdaye.com.geometricweather.daily.adapter.model;

import java.util.TimeZone;

import wangdaye.com.geometricweather.common.basic.models.weather.Astro;
import wangdaye.com.geometricweather.common.basic.models.weather.MoonPhase;
import wangdaye.com.geometricweather.daily.adapter.DailyWeatherAdapter;

public class DailyAstro implements DailyWeatherAdapter.ViewModel {

    private TimeZone timeZone;
    private Astro sun;
    private Astro moon;
    private MoonPhase moonPhase;

    public DailyAstro(TimeZone timeZone, Astro sun, Astro moon, MoonPhase moonPhase) {
        this.timeZone = timeZone;
        this.sun = sun;
        this.moon = moon;
        this.moonPhase = moonPhase;
    }

    public TimeZone getTimeZone() {
        return timeZone;
    }

    public void setTimeZone(TimeZone timeZone) {
        this.timeZone = timeZone;
    }

    public Astro getSun() {
        return sun;
    }

    public void setSun(Astro sun) {
        this.sun = sun;
    }

    public Astro getMoon() {
        return moon;
    }

    public void setMoon(Astro moon) {
        this.moon = moon;
    }

    public MoonPhase getMoonPhase() {
        return moonPhase;
    }

    public void setMoonPhase(MoonPhase moonPhase) {
        this.moonPhase = moonPhase;
    }

    public static boolean isCode(int code) {
        return code == 7;
    }

    @Override
    public int getCode() {
        return 7;
    }
}

File: app/src/main/java/org/breezyweather/daily/adapter/model/DailyAstro.kt
package org.breezyweather.daily.adapter.model

import org.breezyweather.common.basic.models.weather.Astro
import org.breezyweather.common.basic.models.weather.MoonPhase
import org.breezyweather.daily.adapter.DailyWeatherAdapter
import java.util.*

class DailyAstro(
    val timeZone: TimeZone,
    val sun: Astro?,
    val moon: Astro?,
    val moonPhase: MoonPhase?
) : DailyWeatherAdapter.ViewModel {
    override val code = 7

    companion object {
        fun isCode(code: Int) = (code == 7)
    }
}
